nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -446,12 +473,22 @@ Costmap2DROS::mapUpdateLoop(double frequency)
       layered_costmap_->getBounds(&x0, &xn, &y0, &yn);
       costmap_publisher_->updateBounds(x0, xn, y0, yn);
 
+      // TODO(jwallace42): need bounds of each costmap layer?
```

Comment:
Float `bx0_, bxn_, by0_, byn_` are being set from integer bounds at the end of the updating bounds cycle in the `LayeredCostmap::updateMap()`. In this method, all layer's costs are being calculated through `updateCosts()` within this one window (per each processing cycle). And thus, we do not need to worry about each layer's window size during the publishing: it was updated within given by layered costmap `(x0,y0)..(xn,yn)` window: all publishing bounds are supposed to be set to this value `updateBounds(x0, xn, y0, yn)` correctly.